pkg/utils/subst/subst_test.go

```diff
@@ -5,9 +5,6 @@
 package subst
 
 import (
-	"fmt"
```

Comment:
Also note that the tests are using SubstituteByData and SubstituteByValue interchangingly and sometimes they do the same thing and sometimes they don't. Which is super confusing. We should provide a clear way where you can use a struct or not or provide a single function that marshalls a struct if it detects that the given object is not a string or a byte slice.

Reply:
Not sure exactly what you're saying here.    I see that the SubstitutionTarget interface provides these methods
- SubstituteByData(path string, value []byte) error
- SubstituteByValue(path string, value interface{}) error
and I see there are some tests that test one and some tests that test the other.   So long as the interface is providing those two methods it makes sense to me that some tests are testing the one method while some tests are testing the other.

If you are suggesting that the SubstitutionTarget only have 1 method, then I really don't that change should be part of this PR.

Reply:
True. It's not part of this pr.